core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,303 @@
+
+    private final String specification;
+
+    private boolean exponentialConfigured; // base option group. default base option
+    private boolean fixedConfigured;       // base option group.
+    private boolean randomConfigured;      // base option group. only one of base options can be configured.
+    private boolean jitterConfigured;
+    private boolean maxAttemptsConfigured;
+
+    @VisibleForTesting
+    long initialDelayMillis = DEFAULT_INITIAL_DELAY_MILLIS;
+    @VisibleForTesting
+    long maxDelayMillis = DEFAULT_MAX_DELAY_MILLIS;
+    @VisibleForTesting
+    double multiplier = DEFAULT_MULTIPLIER;
+    @VisibleForTesting
+    double minJitterRate = -DEFAULT_MAX_JITTER_RATE;
+    @VisibleForTesting
+    double maxJitterRate = DEFAULT_MAX_JITTER_RATE;
+    @VisibleForTesting
+    int maxAttempts = DEFAULT_MAX_ATTEMPTS;
+    @VisibleForTesting
+    long fixedDelayMillis;
+    @VisibleForTesting
+    long randomMinDelayMillis;
+    @VisibleForTesting
+    long randomMaxDelayMillis;
+
+    /**
+     * Creates a {@link BackoffSpec} from a string.
+     */
+    static BackoffSpec parse(String specification) {
+        requireNonNull(specification, "specification");
+        final BackoffSpec spec = new BackoffSpec(specification);
+        for (String option : OPTION_SPLITTER.split(specification)) {
+            spec.parseOption(option);
+        }
+
+        return spec;
+    }
+
+    private BackoffSpec(String specification) {
+        this.specification = specification;
+    }
+
+    private void parseOption(String option) {
+        if (option.isEmpty()) {
+            return;
+        }
+
+        final List<String> keyAndValues = KEY_VALUE_SPLITTER.splitToList(option);
+        checkArgument(keyAndValues.size() == 2,
+                      "option '%s' (expected 'key=value1:value2...')", option);
+
+        configure(keyAndValues.get(0), keyAndValues.get(1));
+    }
+
+    private void configure(String key, String values) {
+        switch (Ascii.toLowerCase(key)) {
+            case "exponential":
+                exponential(key, values);
+                return;
+            case "fixed":
+                fixed(key, values);
+                return;
+            case "random":
+                randomBackoff(key, values);
+                return;
+            case "jitter":
+                jitter(key, values);
+                return;
+            case "maxattempts":
+                maxAttempts(key, values);
+                return;
+            default:
+                throw new IllegalArgumentException("Unknown key " + key);
+        }
+    }
+
+    private void exponential(String key, String exponentialValues) {
+        checkBaseBackoffConfigured();
+        exponentialConfigured = true;
+
+        final List<String> values = VALUE_SPLITTER.splitToList(exponentialValues);
+        checkValueSize(key, values.size(), 4, exponentialValues);
+
+        initialDelayMillis = parseLong(key, values.get(0), ORDINALS.get(0));
```

Comment:
I'm not sure what's better but what do you think about using the parameter's name such as 'initialDelayMillis' instead of the ordinal?